Systemd example (#890)
lavalink-devs/Lavalink: README.md

- [Server configuration](#server-configuration)
  - [Config](#config)
  - [Binary](#binary)

>     - [Systemd Serivce](#systemd-service)

Reviewer: This seems to be indented a little too much. should it be a sub-chapter of "Binary"?

Author: Does @topi314's thumbs up mean it should be a sub-chapter or it should be moved up a level?

Reviewer 2: it should be on the same page as binary so `###` and not `####`

Author: done

Run with `java -jar Lavalink.jar` from the same directory

> ### Persistence (Unix example)

Reviewer 2: this should be named something like `Systemd Service`
and we should put this under the binary header, so `####`

Author: 1. Added extra hash to "Persistence (Unix Example)"
2. Renamed "Persistence (Unix Example) to "Systemd Service"
3. Changed from ``toml`` to ``<blank>`` for Github
4. Added comments inside of ``.service`` file

Reviewer 2: try `ini` hat should look alright

Run with `java -jar Lavalink.jar` from the same directory

#### Systemd Service

If you want your Lavalink server to run even when the console is closed, you will need to create a `lavalink.service` file inside of your system's directory (commonly located at `/usr/lib/systemd/system`).  Create the file, then add the following components (replacing the text inside "<>" with your values):
 ```ini
[Unit]
# Describe the service
Description=Lavalink Service

# Configure service order
After=syslog.target network.target

[Service]
# The user which will run Lavalink
User=<cen>

# Type of serivce (either simple or forking). DO NOT CHANGE THIS PARAMETER
Type=simple

# Where the program should start
WorkingDirectory=</home/cen/lavalink>

# The command to start Lavalink

> ExecStart=java -jar </home/cen/lavalink>/Lavalink.jar

Reviewer 3: should the ram usage flags be added here too? to mirror the docker file usage

Reviewer 4: Yes

Author: Added.

Reviewer 3: You should add a comment educating users about this flag too as not all systems can have it set as 4G.

Run with `java -jar Lavalink.jar` from the same directory

#### Systemd Service

If you want your Lavalink server to run even when the console is closed, you will need to create a `lavalink.service` file inside of your system's directory (commonly located at `/usr/lib/systemd/system`).  Create the file, then add the following components (replacing the text inside "<>" with your values):
 ```ini
[Unit]
# Describe the service
Description=Lavalink Service

# Configure service order
After=syslog.target network.target

[Service]

> # The user whick will run Lavalink

Reviewer 4: ```suggestion
# The user which will run Lavalink
```

Author: Grammer corrected

User=<cen>

> # Type of serivce
> #   Simple:  Proceed with other jobs once systemd has `fork()`ed it
> #   Forking: The service daemonizes itself and there is no need for systemd for to `fork()` it

Reviewer 4: Referring to forking will only serve to confuse users. The user will probably never want to change this line
```suggestion
```

Author: Fixed.

Run with `java -jar Lavalink.jar` from the same directory

#### Systemd Service

> If you want your Lavalink server to run even when the console is closed, you will need to create a `.service` file inside of your system's directory (commonly located at `/usr/lib/systemd/system`).  Create the file, then add the following components (replacing the text inside "<>" with your values):

Reviewer 2: ```suggestion
If you want your Lavalink server to run even when the console is closed, you will need to create a `lavalink.service` file inside of your system's directory (commonly located at `/usr/lib/systemd/system`).  Create the file, then add the following components (replacing the text inside "<>" with your values):
```

Author: Committed

 ```ini
[Unit]
# Describe the service
Description=Lavalink Service

# Configure service order
After=syslog.target network.target

[Service]

> # Who is using the program
> User=<cen>

Reviewer 2: ```suggestion
# The user as which lavalink should run
User=<cen>
```

Author: Committed

> # The user as which Lavalink should run

Reviewer 4: ```suggestion
# The user which will run Lavalink
```

Author: Changed.

# Type of serivce
#   Simple:  creates a thread that will close on service execution
#   Forking: creates a thread that will continue past service execution

> Type=forking

Author: I'm testing it live rn and I'll get back to you on the results.  I also believe that simple will be the best choice after reading [this article](https://superuser.com/questions/1274901/systemd-forking-vs-simple)

> Type=forking

Author: Can confirm, simple is correct

# Where the program should start

> WorkingDirectory=</home/cen/lavalink>

Reviewer 2: ```suggestion
WorkingDirectory=</home/user/lavalink>
```

Author: Committed

# The command to start Lavalink
ExecStart=java -jar <home/user/lavalink>/Lavalink.jar

> # Restart the service if it closes

Reviewer 4: ```suggestion
# Restart the service if it crashes
```

Author: Changed.

Restart=on-failure

[Install]

> # Start this service ss part of normal system start-up

Reviewer 4: ```suggestion
# Start this service as part of normal system start-up
```

Author: Changed.

WantedBy=multi-user.target
```

> To initiate the service, type 
> ```shell
> sudo systemctl daemon-reload
> sudo systemctl enable lavalink.service
> sudo systemctl start lavalink.service
> ```
> 
> into the console.

Reviewer 4: ```suggestion
To initiate the service, run: 
```shell
sudo systemctl daemon-reload
sudo systemctl enable lavalink.service
sudo systemctl start lavalink.service
```

```

Reviewer 4: GitHub suggestions are quite awkward when they contain markdown code blocks

Author: Changed.

Reviewer 4: I still suggest removing `.service`

Author: Yeah, I've removed the ``.service`` marker

Run with `java -jar Lavalink.jar` from the same directory

#### Systemd Service

If you want your Lavalink server to run even when the console is closed, you will need to create a `lavalink.service` file inside of your system's directory (commonly located at `/usr/lib/systemd/system`).  Create the file, then add the following components (replacing the text inside "<>" with your values):
 ```ini
[Unit]
# Describe the service
Description=Lavalink Service

# Configure service order
After=syslog.target network.target

[Service]
# The user which will run Lavalink
User=<usr>

> 

Reviewer 2: ```suggestion
# The group which will run Lavalink
Group=<usr>
```

# Type of serivce (either simple or forking). DO NOT CHANGE THIS PARAMETER
Type=simple

# Where the program should start
WorkingDirectory=</home/usr/lavalink>

# The command to start Lavalink
ExecStart=java -Xmx4G -jar </home/usr/lavalink>/Lavalink.jar

# Restart the service if it crashes
Restart=on-failure

> 

Reviewer 2: ```suggestion
# Delay each restart by 5s
RestartSec=5s

```

[Install]
# Start this service as part of normal system start-up
WantedBy=multi-user.target
```

To initiate the service, run 
```shell
sudo systemctl daemon-reload
sudo systemctl enable lavalink
sudo systemctl start lavalink
```

> 

Reviewer 2: ```suggestion
To see lavalink logs just run `sudo journalctl -u lavalink` or `tail -f </home/usr/lavalink>/logs/spring.log`
```

Run with `java -jar Lavalink.jar` from the same directory

> #### Systemd Service

Reviewer 2: ```suggestion
### Systemd Service
```

Run with `java -jar Lavalink.jar` from the same directory

#### Systemd Service

> If you want your Lavalink server to run even when the console is closed, you will need to create a `lavalink.service` file inside of your system's directory (commonly located at `/usr/lib/systemd/system`).  Create the file, then add the following components (replacing the text inside "<>" with your values):

Reviewer 4: ```suggestion
If you're using a Systemd-based Linux distribution you may want to install Lavalink as a background service. You will need to create a `lavalink.service` file inside `/usr/lib/systemd/system`. Create the file with the following template (replacing the values inside the `<>` brackets):
```

 ```ini
[Unit]
# Describe the service
Description=Lavalink Service

# Configure service order
After=syslog.target network.target

[Service]
# The user which will run Lavalink
User=<usr>
# The group which will run Lavalink
Group=<usr>

> 
> # Type of serivce (either simple or forking). DO NOT CHANGE THIS PARAMETER

Reviewer 4: ```suggestion
```

Author: Previously it was discussed that we try to comment out everything in the example so a novice user can more clearly understand what each parameter is.  Is there a replacement comment you'd like here instead?

Reviewer 4: That sounds like a miscommunication. Earlier I was suggesting the complete removal of the comment. It is still misleading to present `forking` as an option. Alternatively we could remove the type configuration entirely, as `simple` is apparently the default anyways.

Author: I think removing it entirely is a good option then, I'll update it when I'm back on my home setup

Author: Fixed!

Type=simple

# Where the program should start
WorkingDirectory=</home/usr/lavalink>

# The command to start Lavalink
ExecStart=java -Xmx4G -jar </home/usr/lavalink>/Lavalink.jar

# Restart the service if it crashes
Restart=on-failure
# Delay each restart by 5s
RestartSec=5s

[Install]
# Start this service as part of normal system start-up
WantedBy=multi-user.target
```

To initiate the service, run 
```shell
sudo systemctl daemon-reload
sudo systemctl enable lavalink
sudo systemctl start lavalink
```

> To see lavalink logs just run `sudo journalctl -u lavalink` or `tail -f </home/usr/lavalink>/logs/spring.log`

Reviewer 4: ```suggestion
In addition to the usual log files, you can also view the log with `sudo journalctl -u lavalink`.
```
